core/opendaq/reader/src/typed_reader.cpp

```diff
@@ -301,8 +301,29 @@ SizeT TypedReader<TReadType>::getOffsetToData(const ReaderDomainInfo& domainInfo
             // ss1 << toSysTime(adjusted, domainInfo.epoch, domainInfo.readResolution);
             // std::string s1 = ss1.str();
 
-            if (GreaterEqual<TReadType>::Check(domainInfo.multiplier, static_cast<TReadType>(dataStart[i]), startValue))
+            auto readValue = static_cast<TReadType>(dataStart[i]);
+            if (GreaterEqual<TReadType>::Check(domainInfo.multiplier, readValue, startValue))
             {
+                if (absoluteTimestamp)
+                {
+                    if constexpr (!IsTemplateOf<TReadType, daq::Complex_Number>::value)
+                    {
+                        auto adjustedValue = GreaterEqual<TReadType>::Adjust(readValue, domainInfo.multiplier);  // in domainInfo.maxResolution
+                        auto adjustedValueSysTime = toSysTime(adjustedValue, domainInfo.epoch, domainInfo.resolution);
+                        *absoluteTimestamp = adjustedValueSysTime.time_since_epoch().count();
+                    }
+                    else if constexpr (IsTemplateOf<TReadType, daq::RangeType>::value)
+                    {
+                        auto adjustedValue =
+                            GreaterEqual<TReadType>::Adjust(readValue.start, domainInfo.multiplier);  // in domainInfo.maxResolution
+                        auto adjustedValueSysTime = toSysTime(adjustedValue, domainInfo.epoch, domainInfo.resolution);
+                        *absoluteTimestamp = adjustedValueSysTime.time_since_epoch().count();
+                    }
```

Comment:
Can we ever hit this if statement? Seems like RangeType would be included in the above if.